Yes, on macos 10.13 I get a failure:

```
builder for '/nix/store/qp0k5ncjm3gy1f0cggw2iwk4x2yra6h9-ocaml-cohttp-0.19.3.drv' failed with exit code 2; last 10 log lines:
    "_X509_get_subject_name", referenced from:
        _client_verify_callback in libssl_stubs.a(ssl_stubs.o)
        _ocaml_ssl_get_subject in libssl_stubs.a(ssl_stubs.o)
  ld: symbol(s) not found for architecture x86_64
  clang-7: error: linker command failed with exit code 1 (use -v to see invocation)
  File "caml_startup", line 1:
  Error: Error during linking
  Command exited with code 2.
  E: Failure("Command ''/nix/store/sqr051rii5hmjsqwakqcfs6msyn30297-ocaml-4.02.3/bin/ocamlbuild' lib/cohttp.cma lib/cohttp.cmxa lib/cohttp.a lib/cohttp.cmxs top/cohttp_top.cma top/cohttp_top.cmxa top/cohttp_top.a top/cohttp_top.cmxs lwt-core/cohttp_lwt.cma lwt-core/cohttp_lwt.cmxa lwt-core/cohttp_lwt.a lwt-core/cohttp_lwt.cmxs lwt/cohttp_lwt_unix.cma lwt/cohttp_lwt_unix.cmxa lwt/cohttp_lwt_unix.a lwt/cohttp_lwt_unix.cmxs async/cohttp_async.cma async/cohttp_async.cmxa async/cohttp_async.a async/cohttp_async.cmxs bin/cohttp_server_async.native bin/cohttp_curl_async.native bin/cohttp_server_lwt.native bin/cohttp_proxy_lwt.native bin/cohttp_curl_lwt.native examples/async/hello_world.native examples/async/receive_post.native examples/doc/client_lwt.native examples/doc/server_lwt.native -tag debug -classic-display' terminated with error code 10")
  make: *** [Makefile:20: build] Error 1
```

--- a/pkgs/development/ocaml-modules/ocsigen-server/default.nix
+++ b/pkgs/development/ocaml-modules/ocsigen-server/default.nix
@@ -13,14 +13,14 @@ let mkpath = p: n:
 in
 
 stdenv.mkDerivation rec {
-  version = "2.11.0";
+  version = "2.15.0";
   pname = "ocsigenserver";
 
   src = fetchFromGitHub {
     owner = "ocsigen";
     repo = "ocsigenserver";
     rev = version;
-    sha256 = "0y1ngki7w9s10ip7nj9qb7254bd5sp01xxz16sxyj7l7qz603hy2";
+    sha256 = "15qdkxcbl9c1bbn0fh9awjw0hjn7r6awcn288a9vyxln7icdbifw";
   };
 
   buildInputs = [ which makeWrapper ocaml findlib

--- a/pkgs/development/ocaml-modules/ssl/default.nix
+++ b/pkgs/development/ocaml-modules/ssl/default.nix
@@ -1,29 +1,25 @@
-{ stdenv, fetchzip, which, openssl, ocaml, findlib }:
+{ lib, buildDunePackage, fetchFromGitHub, pkg-config, openssl }:
 
-stdenv.mkDerivation rec {
-  name = "ocaml${ocaml.version}-ssl-${version}";
-  version = "0.5.5";
+buildDunePackage rec {
+  pname = "ssl";
+  version = "0.5.9";
 
-  src = fetchzip {
-    url = "https://github.com/savonet/ocaml-ssl/releases/download/${version}/ocaml-ssl-${version}.tar.gz";
-    sha256 = "0j5zvsx51dg5r7sli7bakv7gfd29z890h0xzi876pg9vywwz9w7l";
+  src = fetchFromGitHub {
+    owner = "savonet";
+    repo = "ocaml-ssl";
+    rev = version;
+    sha256 = "04h02rvzrwp886n5hsx84rnc9b150iggy38g5v1x1rwz3pkdnmf0";
   };
 
-  buildInputs = [which ocaml findlib];
-
+  nativeBuildInputs = [ pkg-config ];
   propagatedBuildInputs = [openssl];
-
-  dontAddPrefix = true;
-
-  createFindlibDestdir = true;
 
   meta = {
     homepage = http://savonet.rastageeks.org/;
     description = "OCaml bindings for libssl ";
     license = "LGPL+link exception";
-    platforms = ocaml.meta.platforms or [];
     maintainers = [
-      stdenv.lib.maintainers.z77z
+      lib.maintainers.z77z
     ];
   };
 }

